### DISCUSSION TITLE
SPI port selection when TFT_eSPI is used as a ESP-IDF component.

### DISCUSSION THREAD

**Discussion by supcik** [Bodmer/TFT_eSPI#2191] [General]
When TFT_eSPI is used as an ESP-IDF component, the files `Kconfig` and `TFT_config.h` are used to configure the device. However, the selection of the SPI port (VSPI or HSPI) seems to be missing.

I suggest adding a _choice_ in the `Kconfig` file:

``` text
choice TFT_SPI_PORT
    prompt "SPI port"
    default TFT_SPI_2_HOST
    help
        The ESP32 has 2 free SPI ports i.e. VSPI (SPI2) and HSPI (SPI3),
        the VSPI is the default. If the VSPI port is in use and pins are
        not accessible (e.g. TTGO T-Beam) then use the HSPI port for the
        TFT display.
    config TFT_VSPI_PORT
        bool "VSPI (SPI2)"
    config TFT_HSPI_PORT
        bool "HSPI (SPI3)"
endchoice
```

and to extend `TFT_config.h` to make use of this configuration:

```cpp
#if CONFIG_TFT_HSPI_PORT
    #define USE_HSPI_PORT
#endif
```

Do you have any objection to doing so? Can I make a _pull request_ for this change?

**Comment by Bodmer** [comment]
A pull request would be welcome as I do not use TFT_eSPI as an ESP-IDF component!

**Reply by xinort** [reply]
wait! TFT_eSPI can be used as ESP-IDF component?!? I'd assumed it was arduino only.

**Reply by supcik** [reply]
Yes it can. It requires that Arduino is also used as a component, but it can. I managed to use it, so feel free to ask for help if you need it.

**Comment by memeticae** [comment]
It also depends on whether you are using the IC alone, or a module like a wrover/wroom
A module usually has psram, eliminating some i/o's and restricting the use of others, especially at startup.
